### DISCUSSION TITLE
Make salt.states.virt consistent with salt.modules.virt and add virt.reset

### DISCUSSION THREAD

**Discussion by ngaugler** [saltstack/salt#67302] [Feature Requests]
`salt.states.virt` has inconsistencies making it difficult to appropriately interface with libvirt.  Within libvirt one can perform both a soft and hard shutdown or reboot/reset of a guest domain.  The module `salt.modules.virt` functions seem to appropriately reflect these features, but the state `salt.states.virt` functions are inconsistent and intermingle terminology.  Documentation is also inconsistent as well adding more confusion.

Below is a mapping showing some basic inconsistencies:

| Type | Task | libvirt | Module | State |
| --- | --- | --- | --- | --- | 
| soft | shutdown | `shutdown` | `salt.modules.virt.shutdown` | `salt.states.virt.stopped` |
| hard | power off | `destroy` | `salt.modules.virt.stop` | `salt.states.virt.powered_off` |
| soft | reboot | `reboot` |  `salt.modules.virt.reboot` | `salt.states.virt.rebooted` |
| hard | reset |  `reset` | `salt.modules.virt.reset` | N/A |

It's inconsistency makes it difficult to anticipate what command will be sent to libvirt. It's so difficult that even the documentation for `salt.states.virt.powered_off` shows the wrong example in the SLS YAML and lists `virt.stopped`  https://docs.saltproject.io/en/latest/ref/states/all/salt.states.virt.html#salt.states.virt.powered_off

It would be easy to add the missing `salt.states.virt.reset` function, but ideally this entire state would be revamped to be consistent with the module.  Also, the documentation should be fixed to reflect the right command.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://join.slack.com/t/saltstackcommunity/shared_invite/zt-1zlfxffs1-NuEH~G9TzOeuNGdsfZIl3w)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by whytewolf** [comment]
Thank you for the report. so. state modules always differ greatly from the execution modules, by design. they are not meant to be one to one comparisons. states are meant to be a description of how the thing should be. it uses the execution modules to get to that state. reset and reboot are actions taken, not states. they should be handled by a mod_watch function. 

take the service state as an example of how this should be. 

there is no service state function for restart or reload. there is a mod_watch that gets triggered when watched states happen. that triggers a restart or reload. 

So there is work to be done. the documentation being one piece, that definitely needs updated. 
and the creation of a mod_watch function to handle reset and reboot.
